bbq/vm/value_deployedcontract.go

```diff
@@ -45,7 +45,7 @@ func init() {
 				var receiver interpreter.Value
 
 				// arg[0] is the receiver. Actual arguments starts from 1.
-				receiver, args = args[ReceiverIndex], args[TypeBoundFunctionArgumentOffset:] // nolint:staticcheck
+				receiver, _ = SplitReceiverAndArgs(context, args)
```

Comment:
The idea in the original code was to still overwrite `args`, even if it is not used, just in case it will be used in the future. It might be good to keep doing it, and even consider doing this everywhere else, to avoid future bugs/misuses of the arguments slice